packages/clay-data-provider/src/useResource.ts

```diff
@@ -181,12 +187,27 @@ const useResource = ({
 					).searchParams.toString()
 				);
 				break;
-			case 'string':
-				promise = fetch(
-					getUrlFormat(link, variables),
-					fetchOptions
-				).then(res => res.json());
+			case 'string': {
+				const url = getUrlFormat(link, variables);
+
+				if (CONCURRENT_PROMISES[url]) {
+					promise = CONCURRENT_PROMISES[url];
+					break;
+				}
+
+				promise = CONCURRENT_PROMISES[url] = fetch(url, fetchOptions)
+					.then(res => {
+						delete CONCURRENT_PROMISES[url];
+
+						return res.json();
+					})
+					.catch(error => {
+						delete CONCURRENT_PROMISES[url];
+
+						throw error;
+					});
```

Comment:
I think we can assume and use `finally()` instead of repeating the `delete` in both `then()` and `catch()` — all [modern browsers support `finally()`](https://caniuse.com/#feat=promise-finally), and core-js (which [we use as  a polyfill for IE in portal](https://github.com/liferay/liferay-portal/blob/c3fe9f2af73a694df4351498ce242f0da18bbe33/modules/apps/frontend-compatibility/frontend-compatibility-ie/build.gradle#L30)) has [supported it for a while](https://github.com/zloirock/core-js/issues/225).

(I think that, as a library, Clay should be following a bring-your-own-polyfill approach [like I said here](https://github.com/liferay/clay/pull/3007#issuecomment-598643306)).

Reply:
Oh yeah, I had forgotten about that method available on `Promise`, thanks!